bumblebee/modules/smartstatus.py

```diff
@@ -0,0 +1,117 @@
+# under Copyright (C) 2015 Marc Herndon and GPL2
+
+"""Displays HDD smart status of different drives or all drives
+
+Parameters:
+    * smartstatus.display: how to display (defaults to "combined", other choices: "seperate" or "singles")
+    * smartstauts.drives: in the case of singles which drives to display, separated comma list value, multiple accepted (defaults to "sda", example:"sda,sdc")
+"""
+
+import os
+
+import bumblebee.util
+import bumblebee.output
+import bumblebee.engine
+
+from shutil import which
+from subprocess import Popen, PIPE
+
+
+class Module(bumblebee.engine.Module):
+    def __init__(self, engine, config):
+        super(Module, self).__init__(engine, config, None)
+        self.devices = self.list_devices()
+        self.display = self.parameter('display', 'combined')
+        self.drives = self.parameter('drives', 'sda')
+        self.widgets(self.create_widgets())
+
+    def create_widgets(self):
+        widgets = []
+        if self.display == 'combined':
+            widget = bumblebee.output.Widget()
+            widget.set('device', 'combined')
+            widget.set('assessment', self.combined())
+            self.output(widget)
+            widgets.append(widget)
+        else:
+            for device in self.devices:
+                if self.display == "singles" and device not in self.drives:
+                    continue
+                widget = bumblebee.output.Widget()
+                widget.set('device', device)
+                widget.set('assessment', self.smart(device))
+                self.output(widget)
+                widgets.append(widget)
+        return widgets
+
+    def update(self, widgets):
+        for widget in widgets:
+            device = widget.get('device')
+            if device == 'combined':
+                widget.set('assessment', self.combined())
+                self.output(widget)
+            else:
+                widget.set('assessment', self.smart(device))
+                self.output(widget)
+
+    def output(self, widget):
+        device = widget.get('device')
+        assessment = widget.get('assessment')
+        widget.full_text("{}: {}".format(device, assessment))
+
+    def state(self, widget):
+        states = []
+        assessment = widget.get('assessment')
+        if assessment == 'Pre-fail':
+            states.append('warning')
+        if assessment == 'Fail':
+            states.append('critical')
+        return states
+
+    def combined(self):
+        for device in self.devices:
+            result = self.smart(device)
+            if result == 'Fail':
+                return 'Fail'
+            if result == 'Pre-fail':
+                return 'Pre-fail'
+        return 'OK'
+
+    def list_devices(self):
+        for (root, folders, files) in os.walk('/dev'):
+            if root == '/dev':
+                devices = {"".join(filter(lambda i: i.isdigit() == False, file)) for file in files if 'sd' in file}
```

Comment:
For a future extension, it might be nice to support also other naming schemes than "sdX" here? I have an NVMe, for instance, so my root disk is called /dev/nvme0n1

Reply:
As I mentionned in the first line, the smart function was inpired from py-smart with that line specifically being more or less a copy with the exception of sudo that I added, I did see that in there and thought about it and remembered that was a disk format, it completely missed my attention when I was created the list devices as I only have sdX devices, is there any other devices I should be paying attention to?

Reply:
Could you let me know what you have to enter to get the smart status of your drive?
 `sudo smartctl -a /dev/XXXX`

I mean it wont take much for me to do, I just need to know the path of the devices to watch out for, as I currently do not have one. I have no idea.

Reply:
@martindoublem I use `sudo smartctl -a /dev/nvme0n1`